core/market/src/db/dao/negotiation_events.rs

```diff
@@ -79,7 +80,7 @@ impl<'c> NegotiationEventsDao<'c> {
 
             let events = dsl::market_negotiation_event
                 .filter(dsl::subscription_id.eq(&subscription_id))
-                .order_by(dsl::timestamp.asc())
+                .order_by(sql::<sql_types::Bool>("RANDOM()"))
```

Comment:
kinda controversial, because we will be hiding `AgreementEvent`s, which should have priority.
We should only shuffle `ProposalEvent`s...

Reply:
We decided to create issue for later:  #932